pathauto.install

```diff
@@ -51,3 +52,137 @@ function pathauto_update_8001() {
+          continue;
+        }
+
+        if (isset($bundle_info[$bundle])) {
+          // This is a pattern for a bundle, such as "node_article".
+          $entity = entity_create('pathauto_pattern', [
+            'id' => $entity_type . '_' . $bundle,
+            'label' => $entity_label . ' ' . $bundle_info[$bundle]['label'],
+            'type' => 'canonical_entities:' . $entity_type,
+            'pattern' => $bundle_patterns['default'],
+            'weight' => -5,
+          ]);
+
+          // Add the bundle condition.
+          $entity->addSelectionCondition([
+            'id' => 'entity_bundle:' . $entity_type,
+            'bundles' => array($bundle),
+            'negate' => FALSE,
+            'context_mapping' => [ $entity_type => $entity_type ],
+          ]);
+
+          $entity->save();
+        }
+        else {
+          // This is either a language dependent pattern such as "article_es" or
+          // an unknown bundle or langcode. Let's figure it out.
+          $matches = NULL;
+          $langcode = NULL;
+          preg_match('/^(.*)_([a-z-]*)$/', $bundle, $matches);
+          if (count($matches) == 3) {
+            list(, $extracted_bundle, $langcode) = $matches;
+            $language = $language_manager->getLanguage($langcode);
+          }
+          // Validate bundle, langcode and language.
+          if (!isset($bundle_info[$extracted_bundle]) || ($langcode == NULL) || ($language == NULL)) {
+            $messages[] = t('Unrecognized entity bundle @entity:@bundle was not processed. It defines the following patterns: @patterns', array(
+              '@entity' => $entity_type,
+              '@bundle' => $bundle,
+              '@patterns' => print_r($entity_patterns, TRUE),
+            ));
+            continue;
+          }
+
+          // This is a pattern for a bundle and a language, such as "node_article_es".
+          $entity = entity_create('pathauto_pattern', [
+            'id' => $entity_type . '_' . $extracted_bundle . '_' . $langcode,
+            'label' => $entity_label . ' ' . $bundle_info[$extracted_bundle]['label'] . ' ' . $language->getName(),
+            'type' => 'canonical_entities:' . $entity_type,
+            'pattern' => $bundle_patterns['default'],
+            'weight' => -10,
+          ]);
+
+          // Add the bundle condition.
+          $entity->addSelectionCondition([
+            'id' => 'entity_bundle:' . $entity_type,
+            'bundles' => array($extracted_bundle),
+            'negate' => FALSE,
+            'context_mapping' => [ $entity_type => $entity_type ],
+          ]);
+
+          // Add the language condition.
+          $entity->addSelectionCondition([
+            'id' => 'language',
+            'langcodes' => [ $langcode => $langcode ],
+            'negate' => FALSE,
+            'context_mapping' => [
+              'language' => $entity_type . ':' . $entity_type_manager->getDefinition($entity_type)->getKey('langcode') . ':language',
```

Comment:
@Berdir isn't this doing the same as the follwing?

```
          $new_definition = new ContextDefinition('language', 'Language');
          $new_context = new Context($new_definition);
          $pattern->addContext($language_context, $new_context);
```

I will test creating a pattern through the interface anyway to see if there is a difference in the resulting config entity.